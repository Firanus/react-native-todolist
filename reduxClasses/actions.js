export const types = {
  ADD_ITEM: 'ADD_ITEM',
  REMOVE_COMPLETED_ITEMS: 'REMOVE_COMPLETED_ITEMS',
  REMOVE_ITEM: 'REMOVE_ITEM',
  TOGGLE_COMPLETED: 'TOGGLE_COMPLETED'
};

export const actionCreators = {
  add: (itemTitle) => {
    return {type: types.ADD_ITEM, payload: itemTitle}
  },
  removeCompleted: () => {
    return {type: types.REMOVE_COMPLETED_ITEMS}
  }
};