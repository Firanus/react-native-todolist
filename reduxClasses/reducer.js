import {types} from './actions'

const initialState = {
  todos: [ {id: 0, title: 'defaultItem', isCompleted: false} ]
};

export const reducer = (state = initialState, action) => {
  const { todos } = state;
  const { type, payload } = action;

  switch(type){
    case types.ADD_ITEM:
      return {
        ...state,
        todos: [
          {
            id: state.todos.reduce((maxId, elem) => elem.id > maxId ? elem.id : maxId, 0) + 1,
            title: payload,
            isCompleted: false
          },
          ...state.todos],
      };
    case types.REMOVE_COMPLETED_ITEMS:
      return {
        ...state,
        todos: state.todos.filter((todo) => {
          return todo.isCompleted = false
        }),
      };
      break;
    default:
      break;
  }

  return state;
};