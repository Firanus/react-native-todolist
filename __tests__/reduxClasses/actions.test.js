import { types, actionCreators } from '../../reduxClasses/actions'

test('Action types are correct', () => {
  expect(types).toMatchSnapshot();
});

describe('The actionCreators object', () => {
  test('creates an add action', () => {
    const itemTitle = 'My Test Task';
    expect(actionCreators.add(itemTitle)).toMatchSnapshot();
  });

  test('creates a removeCompleted action', () => {
    expect(actionCreators.removeCompleted()).toMatchSnapshot();
  })
});
