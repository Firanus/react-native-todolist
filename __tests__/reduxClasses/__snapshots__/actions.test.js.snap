// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Action types are correct 1`] = `
Object {
  "ADD_ITEM": "ADD_ITEM",
  "REMOVE_COMPLETED_ITEMS": "REMOVE_COMPLETED_ITEMS",
  "REMOVE_ITEM": "REMOVE_ITEM",
  "TOGGLE_COMPLETED": "TOGGLE_COMPLETED",
}
`;

exports[`The actionCreators object creates a removeCompleted action 1`] = `
Object {
  "type": "REMOVE_COMPLETED_ITEMS",
}
`;

exports[`The actionCreators object creates an add action 1`] = `
Object {
  "payload": "My Test Task",
  "type": "ADD_ITEM",
}
`;
